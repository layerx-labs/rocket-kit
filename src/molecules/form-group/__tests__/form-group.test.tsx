import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { FormGroup, TextField } from '../../..';

describe('FormGroup', () => {
  it('renders', () => {
    const { asFragment } = render(
      <FormGroup label="label">
        <TextField type="text" value="this is my text" />
      </FormGroup>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct value', () => {
    const textValue = 'this is my text';
    render(
      <FormGroup label="label">
        <TextField type="text" value={textValue} />
      </FormGroup>
    );
    const foundInput = screen.getByDisplayValue(textValue);
    expect(foundInput).toHaveValue(textValue);
  });
});
