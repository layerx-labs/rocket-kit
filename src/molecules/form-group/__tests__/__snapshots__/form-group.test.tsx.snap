// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup renders 1`] = `
<DocumentFragment>
  .c2 {
  border-width: 0.0625rem;
  border-style: solid;
  border-color: #D9D4ED;
  border-radius: 0.5rem;
  background-repeat: no-repeat;
  background-size: 1.25rem 1.25rem;
  background-position: 0.625rem 50%;
  height: 3.125rem;
  padding: 0.625rem;
  font-family: inherit;
  font-size: 16px;
  -webkit-transition-property: border-color;
  transition-property: border-color;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c2:hover {
  border-color: #C7BFE4;
}

.c2:focus {
  border-color: #4329A6;
  outline: none;
}

.c2:disabled {
  border-color: #D9D4ED;
  background-color: #E3E3E8;
  color: #ABACBA;
}

.c2:invalid {
  border-color: #EB5757;
  outline: none;
  box-shadow: none;
  -webkit-box-shadow: none;
  -moz-box-shadow: none;
}

.c2:required {
  border-color: #D9D4ED;
}

.c2::-webkit-input-placeholder,
.c2:-ms-input-placeholder,
.c2::-ms-input-placeholder {
  color: #C7C8D1;
  opacity: 1;
}

.c2::-moz-placeholder,
.c2:-ms-input-placeholder,
.c2::-ms-input-placeholder {
  color: #C7C8D1;
  opacity: 1;
}

.c2:-ms-input-placeholder,
.c2:-ms-input-placeholder,
.c2::-ms-input-placeholder {
  color: #C7C8D1;
  opacity: 1;
}

.c2::placeholder,
.c2:-ms-input-placeholder,
.c2::-ms-input-placeholder {
  color: #C7C8D1;
  opacity: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.75rem;
  font-weight: 600;
  color: #C7C8D1;
}

.c1 svg {
  margin-left: 0.3125rem;
  width: auto;
  height: 1rem;
}

.c0 {
  display: inline-block;
}

.c0 > label {
  margin-bottom: 0.3125rem;
  color: #C7C8D1;
}

.c0 > label span {
  color: #000000;
}

.c0:not(:last-child) {
  margin-bottom: 0.9375rem;
}

<div
    class="c0"
  >
    <label
      class="c1 label"
    >
      label
    </label>
    <input
      class="c2 text-field"
      placeholder=""
      type="text"
      value="this is my text"
    />
  </div>
</DocumentFragment>
`;
